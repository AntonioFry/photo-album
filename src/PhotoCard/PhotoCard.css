.photo-card {
  align-items: flex-end;
  background-color: rgba(247, 247, 247, 0.5);
  border-radius: 10px;
  box-shadow: 10px 10px 19px -3px rgba(0,0,0,0.5);
  display: flex;
  height: 266.66px;
  margin: 25px auto;
  width: 400px;
}

.photo-card:hover .photo-comment-container {
  visibility: visible;
}

.taken-photo {
  border-radius: 10px;
  height: 100%;
  width: 100%;
}

.photo-comment-container {
  visibility: hidden;
  background-color: rgba(247, 247, 247, 0.5);
  border-bottom-left-radius: 10px;
  border-bottom-right-radius: 10px;
  padding: 5px;
  width: 400px;
}

.photo-comment {
  font-size: 13px;
  margin: 0px;
}

.photo-card-btn-container {
  display: flex;
  justify-content: flex-end;
  width: 100%;
}

.album-name-btn {
  background-color: transparent;
  border: none;
  cursor: pointer;
  font-size: 15px;
  font-weight: bold;
  margin: 3px 5px;
  text-align: right;
}