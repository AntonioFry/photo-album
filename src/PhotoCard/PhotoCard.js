import React from 'react';

const PhotoCard = ({ photoDescription, imageSrc, photoComment }) => {
  return (
    <article>
      <img src={''} />
      <p></p>
    </article>
  )
}