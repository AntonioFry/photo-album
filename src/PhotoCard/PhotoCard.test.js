import React from 'react';
import ReactDOM from 'react-dom';
import PhotoCard from './PhotoCard';
import renderer from 'react-test-renderer';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('PhotoCard', () => {

  afterEach(cleanup);
  
  it('Should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<PhotoCard imageSrc='some.JPG' photoComment="It's something" testId="photo-comment"/>, div);
  });

  it('Should render correctly', () => {
    const { getByTestId } = render(<PhotoCard imageSrc='some.JPG' photoComment="It's a photo of something" photoDescription='A something' testId="photo-comment"/>);
    expect(getByTestId('photo-comment')).toHaveTextContent("It's a photo of something");
  });

  it('Should match snapshot', () => {
    const tree = renderer.create(<PhotoCard imageSrc='some.JPG' photoComment="It's something" testId="photo-comment"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});