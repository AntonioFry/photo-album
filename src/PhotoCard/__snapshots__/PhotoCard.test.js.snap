// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoCard Should match snapshot 1`] = `
<article
  className="photo-card"
  style={
    Object {
      "backgroundImage": "url(some.JPG)",
      "backgroundSize": "cover",
    }
  }
>
  <div
    className="photo-comment-container"
  >
    <p
      className="photo-comment"
      data-testid="photo-comment"
    >
      It's something
    </p>
    <div
      className="photo-card-btn-container"
    >
      <button
        className="album-name-btn"
        onClick={[Function]}
      />
    </div>
  </div>
</article>
`;
